Upstream: yes
Reason: fix crash in depending applications, see
http://mail.kde.org/pipermail/release-team/2014-October/008085.html
for details

From 3576df163c8c64ebba74fc51c95de900dbf0220e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Gr=C3=A4=C3=9Flin?= <mgraesslin@kde.org>
Date: Thu, 9 Oct 2014 11:06:54 +0200
Subject: [PATCH] Fix incorrect count of netwm atoms

This fixes a regression introduced with
59cb063b99e6ec13c0bc3cf3fd95a01258184e86.

The commit added 4 new atoms but only incremented the atom count by 3.
Which can cause a crash as soon as the NET classes are used.

To ensure that such an error is not introduced again, an assert is
added.

BUG: 339771
REVIEW: 120539
---
 src/netwm.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/netwm.cpp b/src/netwm.cpp
index 3107a23..7b398d4 100644
--- a/src/netwm.cpp
+++ b/src/netwm.cpp
@@ -386,7 +386,7 @@ static QByteArray get_atom_name(xcb_connection_t *c, xcb_atom_t atom)
 }
 #endif
 
-static const int netAtomCount = 91;
+static const int netAtomCount = 92;
 
 static void create_netwm_atoms(xcb_connection_t *c)
 {
@@ -498,6 +498,7 @@ static void create_netwm_atoms(xcb_connection_t *c)
     };
 
     assert(!netwm_atoms_created);
+    assert(sizeof(atoms)/sizeof(atoms[0]) == netAtomCount + 1);
 
     // Send the intern atom requests
     xcb_intern_atom_cookie_t cookies[netAtomCount];
-- 
2.1.2

